tools/base/utils.py

```diff
@@ -26,3 +27,42 @@ def coverage_with_data_file(data_file: str) -> Iterator[str]:
         with open(tmprc, "w") as f:
             parser.write(f)
         yield tmprc
+
+
+class BufferUtilError(Exception):
+    pass
+
+
+@contextmanager
+def nested(*contexts):
+    with ExitStack() as stack:
+        yield [stack.enter_context(context) for context in contexts]
+
+
+@contextmanager
+def buffered(stdout: list = None, stderr: list = None, mangle=None) -> None:
```

Comment:
wrt to this util - the stdlib context managers for capturing stdout/err use a streaming interface - which ive tried to abstract away here to make it easier to re/use (and ultimately cut code)

Reply:
mangle needs `type`